xlml/utils/xpk.py

```diff
@@ -37,98 +40,122 @@ def generate_workload_id(benchmark_id: str) -> str:
   return f"{short_benchmark}{short_id}"
 
 
-def _run_workload(
+@task
+def run_workload(
     task_id: str,
     cluster_project: str,
     zone: str,
     cluster_name: str,
     benchmark_id: str,
     workload_id: str,
+    gcs_path: str,
     docker_image: str,
     accelerator_type: str,
     run_cmds: str,
     num_slices: int = 1,
 ):
   """Run workload through xpk tool."""
-  import subprocess
+  from subprocess import run, STDOUT, PIPE
+  from xlml.apis import metric_config
+
+  run_cmds = f"export {metric_config.SshEnvVars.GCS_OUTPUT.name}={gcs_path}; {run_cmds}"
 
   cmds = (
-      "set -xue",
-      "git clone https://github.com/google/xpk.git /tmp/xpk",
-      "cd /tmp/xpk",
-      "gcloud components install kubectl",
-      f"gcloud config set project {cluster_project}",
-      f"gcloud config set compute/zone {zone}",
+      "set -xu",
+      "TMPDIR=$(mktemp -d)",
+      "export KUBECONFIG=${TMPDIR}/xpk.conf",
+      "git clone https://github.com/google/xpk ${TMPDIR}/xpk",
       (
-          "python3 xpk.py workload create"
+          "python ${TMPDIR}/xpk/xpk.py workload create"
           f" --cluster={cluster_name} --workload={workload_id}"
           f" --command='{run_cmds}' --device-type={accelerator_type}"
           f" --num-slices={num_slices} --docker-image={docker_image}"
+          f" --project={cluster_project} --zone={zone}"
       ),
+      "rc=$?",
+      "rm -rf $TMPDIR",
+      "exit $rc",
   )
 
-  subprocess.run(["bash", "-c", ";".join(cmds)], check=True)
+  res = run(["bash", "-c", ";".join(cmds)], stderr=STDOUT, stdout=PIPE)
```

Comment:
`SubprocessHook` should work now since you're not in a container: https://airflow.apache.org/docs/apache-airflow/stable/_api/airflow/hooks/subprocess/index.html#airflow.hooks.subprocess.SubprocessHook

You can pass in the env as a dict instead of tacking on `export`s, and the logs get plumbed automatically.